sunpy/net/helioviewer.py

```diff
@@ -75,11 +104,23 @@ def get_closest_image(self, date, **kwargs):
         >>> print(metadata['date'])  # doctest: +REMOTE_DATA
         2012-01-01 00:00:07
         """
+
+        if('observatory' in kwargs):
+            observatory = kwargs['observatory'] if(kwargs['observatory']) else 'None' 
+            instrument = kwargs['instrument'] if(kwargs['instrument']) else 'None'
+            detector = kwargs['detector'] if(kwargs['detector']) else 'None'
+            measurement = kwargs['measurement'] if(kwargs['measurement']) else 'None'
+            key = (observatory, instrument, detector, measurement)
+            sourceid = self._source_dict[key]
+
+        elif('sourceId' in kwargs):
+            sourceid = kwargs['sourceId']
```

Comment:
Do we want to have a nice error message sourceid is not found?

Reply:
We can. Shall we add one?

Reply:
I would think so.